kythe/go/platform/tools/kzip/stripcmd/stripcmd.go

```diff
@@ -0,0 +1,154 @@
+package stripcmd // import "kythe.io/kythe/go/platform/tools/kzip/stripcmd"
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"path/filepath"
+
+	"kythe.io/kythe/go/platform/kzip"
+	"kythe.io/kythe/go/platform/tools/kzip/flags"
+	"kythe.io/kythe/go/platform/vfs"
+	"kythe.io/kythe/go/util/cmdutil"
+
+	"bitbucket.org/creachadair/stringset"
+	"github.com/google/subcommands"
+)
+
+type stripCommand struct {
+	cmdutil.Info
+
+	input    string
+	output   string
+	append   bool
+	encoding flags.EncodingFlag
+}
+
+// New creates a new subcommand for merging kzip files.
+func New() subcommands.Command {
+	return &stripCommand{
+		Info:     cmdutil.NewInfo("strip", "strip units from kzip file", "--input path --output path unit-hash*"),
+		encoding: flags.EncodingFlag{Encoding: kzip.EncodingJSON},
+	}
+}
+
+// SetFlags implements the subcommands interface and provides command-specific flags
+// for merging kzip files.
+func (c *stripCommand) SetFlags(fs *flag.FlagSet) {
+	fs.StringVar(&c.output, "output", "", "Path to output kzip file")
+	fs.StringVar(&c.input, "input", "", "Path to input kzip file")
+	fs.Var(&c.encoding, "encoding", "Encoding to use on output, one of JSON, PROTO, or ALL")
+}
+
+// Execute implements the subcommands interface and strips the input file.
+func (c *stripCommand) Execute(ctx context.Context, fs *flag.FlagSet, _ ...interface{}) subcommands.ExitStatus {
+	if c.output == "" {
+		return c.Fail("required --output path missing")
+	}
+	if c.input == "" {
+		return c.Fail("required --input path missing")
+	}
+	opt := kzip.WithEncoding(c.encoding.Encoding)
+	dir, file := filepath.Split(c.output)
+	if dir == "" {
+		dir = "."
+	}
+	tmpOut, err := vfs.CreateTempFile(ctx, dir, file)
+	tmpName := tmpOut.Name()
+	defer func() {
+		if tmpOut != nil {
+			tmpOut.Close()
+			vfs.Remove(ctx, tmpName)
+		}
+	}()
+	if err != nil {
+		return c.Fail("Error creating temp output: %v", err)
+	}
+	units := stringset.FromKeys(fs.Args())
+	if err := stripArchive(ctx, tmpOut, c.input, units, opt); err != nil {
+		return c.Fail("Error stripping archives: %v", err)
+	}
+	if err := vfs.Rename(ctx, tmpName, c.output); err != nil {
+		return c.Fail("Error renaming tmp to output: %v", err)
+	}
+	return subcommands.ExitSuccess
+}
+
+func stripArchive(ctx context.Context, out io.WriteCloser, input string, digests stringset.Set, opts ...kzip.WriterOption) error {
+	filesAdded := stringset.New()
+
+	f, err := vfs.Open(ctx, input)
+	if err != nil {
+		return fmt.Errorf("error opening archive: %v", err)
+	}
+	defer f.Close()
+
+	stat, err := vfs.Stat(ctx, input)
+	if err != nil {
+		return err
+	}
+	size := stat.Size()
+	if size == 0 {
+		log.Printf("Skipping empty .kzip: %s", input)
+		return nil
+	}
+
+	rd, err := kzip.NewReader(f, size)
+	if err != nil {
+		return fmt.Errorf("error creating reader: %v", err)
+	}
+
+	wr, err := kzip.NewWriteCloser(out, opts...)
+	if err != nil {
+		out.Close()
```

Comment:
Why is `out` closed here but none of the other error paths above?  Maybe just pass in a `io.Writer` and handle the close upstream.

Reply:
Cleaned up; out was already being closed in caller. defer()red the wr.Close() call.